File: iosApp/iosApp/analytics/AnalyticsImpl.swift
//
//  AnalyticsImpl.swift
//  iosApp
//
//  Created by Alfa on 25.06.2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import shared
import FirebaseAnalytics

class AnalyticsImpl : BaseAnalytics {
    
    override init() {
        super.init()
    }
    
    override func sendEventToServer(event: String, params: [String : Any]) {
        Analytics.logEvent(event, parameters: params)
    }
}

File: androidApp/src/main/java/com/example/kmpapp/android/analytics/AnalyticsImpl.kt
package com.example.kmpapp.android.analytics

import com.example.core.analytics.BaseAnalytics
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.logEvent

class AnalyticsImpl(
    private val firebaseAnalytics: FirebaseAnalytics
) : BaseAnalytics() {

    override fun sendEventToServer(event: String, params: Map<String, String?>) {
        firebaseAnalytics.logEvent(event) {
            params.forEach { (k, v) ->
                param(k, v.toString())
            }
        }
    }
}